### PR TITLE
refactor(bin-designer): tighten the lip-only lid feature after review

### PR DESCRIPTION
**Pull request by andymai**
Post-merge polish on #2934 / #2936. **No geometry behaviour change** — the generated solids are identical.

## Tests that didn't test what they claimed

Four of the tests I shipped would have passed with the feature broken. Each fix below was mutation-tested (break the code, confirm the test fails):

| Test | Passed with the feature broken because | Now |
|---|---|---|
| "fits text in the grid frame" | asserted only `triangleCount` changed — true with the grid-frame branch deleted | locates glyph recesses by their floor Z and asserts the midpoint is on the grid, not at `outerOffsetX`. Mutation: midpoint moves to **3.72**, test fails |
| "follows the polygon outline" | asserted a bounding box, which is set by the *slab* and blind to the pocket | probes for lip material along the L's concave edge. Mutation (rectangular cutter): **0 vertices**, test fails |
| "splits off as a valid glue-on plate" | accepted `maxZ` in 4..6 — the full grid's plate is 5.0, lip-only 4.5 | pinned to the knife edge |
| overhang lip probe | `lipEdgeMinX` returns `Infinity` when nothing matches, and `toBeCloseTo(Infinity, Infinity)` passes | guarded with a finite check |

Plus new emboss and through-cut coverage on the lip floor — both are genuinely new geometry paths (emboss fuses glyphs into the pocket cavity after the slab is fused; through-cut pierces the floor into the mating cavity).

## A comment asserting a constraint that doesn't exist

I had written that `POCKET_PROFILE` must stay in descending order "because reversing a ruled loft flips the solid's face orientation." I never verified it. Reversing the array leaves all 73 tests passing, including triangle counts and bounding boxes. Deleted — an unverified constraint like that blocks a legitimate refactor forever.

## Two user-visible gaps

**The toggle is a no-op on a single-cell lid.** A 1x1 grid is already one pocket, so lip-only emits identical geometry — my own test asserts exactly that. 1x1 is the most common Gridfinity size, so silently doing nothing there is the common case, not an edge case. Now says so.

**Every stackable lid exports flipped** (`keepsNaturalOrientation` only spares trays). That lands a lip-only top pockets-down, where the full grid's interior ridges gave the floor per-cell bridges (~36mm spans) and lip-only gives one unsupported span across the whole opening. The hint now points at the separate baseplate, which is the existing fix — and which the `separateStackPlate` docs already describe for exactly this reason.

## Also

- Normalise `stackLipOnly` when *entering* the stackable surface. It was cleared on exit but not on entry, so an imported design carrying `{stackableTop: false, stackLipOnly: true}` would land on lip-only unannounced.
- Fixed a `useCallback` dep-array omission on `setTopSurface` — with React Compiler's `preserve-manual-memoization` this was a lint error *and* a stale-closure bug (re-selecting "Stackable" would have cleared lip-only).
- Two doc comments still described the pre-#2936 text gate ("stackable top" rather than "full stack grid"), including `SurfaceTextConfig.lidText`, which is the type a caller reads first.
- Two pre-existing comments said the magnet/baseplate switches are "disabled" when they're hidden (both sit inside `topSurface === 'stackable' &&`).
- Net −56 comment lines: dropped restatement, a "byte-identically" claim nothing verifies, and a "first config where..." superlative that rots silently.

## Investigated and rejected

- **Prototype-caching the per-cell pocket cutters.** A 6x4 grid builds 24 identical 7-section lofts. Caching one per distinct cell size gave byte-identical output (8812 triangles) and **zero speedup** — the cost is the boolean and tessellation, not the lofts. Reverted rather than carry a `Map` and prototype lifecycle for nothing.
- **A magnet-pocket keep-out for lid text.** The engrave clamp (`hostThickness - LID_TEXT_ENGRAVE_FLOOR`) already guarantees ≥0.4mm of plate remains regardless of pocket depth, so a glyph crossing a pocket cannot breach into the mating cavity.

## Verification

`pnpm run quality` clean (0 errors; only pre-existing `max-lines` warnings). 4468 unit tests across `src/features/bin-designer` + `api/lib` + `lidBuilder`, 75 lid scenario tests against real OCCT. The e2e visual spec was **actually executed** this time (it passes in 15.8s) — worth noting that it and its siblings only run manually; CI runs `playwright.smoke.config.ts`, not the full `e2e/` suite.

---
## Summary by cubic
Polish the lip-only stacking lid feature: clearer UI and consistent state handling, plus stronger geometry tests. No change to generated geometry.

- Bug Fixes
  - UI: Show “No effect at this size” for 1×1 lids. For multi-cell lip-only, show a print note pointing to the separate baseplate and hide it once split.
  - State: Normalize `stackLipOnly` when switching to stackable so imports don’t land on lip-only silently; hide gated switches; fix `setTopSurface` dependency array.
  - Text: Allow lid text on the lip-only recessed floor; keep it disabled for full stack grid and polygon lids.
  - Generation: Add a minimum dimension floor for pocket outlines; remove an incorrect loft-order constraint; update types/docs for the new behavior.
  - i18n: Add localized strings for the new no-op and print-note hints; correct the Czech term for the print note.

- Tests
  - Replace weak checks with direct probes: detect lip along concave mask edges; locate glyph floors and assert their midpoint is grid-anchored (not `outerOffsetX`); guard Infinity cases.
  - Pin split baseplate Z to the lip’s knife edge.
  - Add emboss and through-cut coverage on the lip floor; verify overlap with magnet pockets.
  - Add UI tests for 1×1 no-op, print note visibility with/without split plate, and import normalization.

<sup>Written for commit f3da7aa505b645a961314a08d5a4cc4d9d35c639. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/andymai/gridfinity-layout-tool/pull/2938?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: src/i18n/locales/cs.json [hunk 1]
```diff
@@ -857,6 +857,8 @@
   "binDesigner.lid.separateStackPlateHint": "Základní deska se tiskne naplocho samostatně (mřížka nahoru, lepicí plocha dolů), aby víko zůstalo bez podpor. Po tisku ji nalep na horní plochu víka.",
   "binDesigner.lid.stackGridHint": "Plná mřížka zásuvek přes celou horní plochu. Boxy postavené na víko zapadnou do jednotlivých buněk.",
   "binDesigner.lid.stackLipOnly": "Pouze stohovací lem",
+  "binDesigner.lid.stackLipOnlyNoOp": "V této velikosti nemá efekt — jednobuňková horní plocha je už jedna kapsa.",
+  "binDesigner.lid.stackLipOnlyPrintNote": "Víko se tiskne dnem vzhůru, takže lem-only nechá celou plochu bez podpory. Samostatná základní deska níže se tomu vyhne.",
```
[the remark this answers is not in the document]
Comment (andymai) [reply src/i18n/locales/cs.json]: Fixed — "lem-only" was a half-English hybrid I introduced. Now "samotný lem", matching the term the feature label already uses ("Pouze stohovací lem").

Audited the other 13 locales for the same mistake while I was in there. Spanish had a milder version — "un borde solo" reads awkwardly, so it is now "solo el borde". Also grepped the ja/ko/zh-CN/uk strings for stray Latin-script words: none.

— claude
